http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/ClusterHelper.scala

```diff
@@ -59,45 +62,39 @@ class ClusterHelper(
       // it's ok to re-add the roles.
       IO.fromFuture(IO(dbRef.inTransaction { _.clusterQuery.countActiveByClusterServiceAccount(email) })).flatMap {
         count =>
-          if (count > 0 && create == false) {
+          if (count > 0 && !createCluster) {
             IO.unit
           } else {
             retryIam(googleProject, email, Set("roles/dataproc.worker"))
           }
       }
     } getOrElse IO.unit
 
-    // TODO: replace this logic with a group based approach so we don't have to manipulate IAM directly in the image project.
-    // See https://broadworkbench.atlassian.net/browse/IA-1364
-    //
-    // Add the Compute Image User role in the image project to the Google API service account.
-    // This is needed in order to use a custom dataproc VM image.
-    // If a custom image is not being used, this is not necessary.
+    // TODO Template and get the group and domain values from config
+    val dataprocImageUserGoogleGroupName = "kyuksel-test-dataproc-image-group-11"
+    val dataprocImageUserGoogleGroupEmail = WorkbenchEmail(s"$dataprocImageUserGoogleGroupName@test.firecloud.org")
+
+    // Add the user's service account to the Google group that has compute.imageUser role on the custom Dataproc image project.
     val computeImageUserIO = dataprocConfig.customDataprocImage.flatMap(parseImageProject) match {
```

Comment:
Maybe we should split the computeImageUser group stuff to a new method because it's not really manipulating IAM anymore.